File: MapItem.php
<?php
namespace ffan\dop;

/**
 * Class MapItem
 * @package ffan\dop
 */
class MapItem extends Item
{
    /**
     * @var Item 键类型
     */
    private $key_type;

    /**
     * @var Item 值类型
     */
    private $value_type;

    /**
     * @var int 类型
     */
    protected $type = ItemType::MAP;

    /**
     * 设置键类型
     * @param Item $key_type
     * @throws DOPException
     */
    public function setKeyType(Item $key_type)
    {
        //目前只支持int 和 string类型的key
        if($key_type !== ItemType::INT && $key_type !== ItemType::STRING) {
            throw new DOPException('key type error. '. $this->getDocInfo());
        }
        $this->key_type = $key_type;
    }

    /**
     * 设置值类型
     * @param Item $value_type
     */
    public function setValueType(Item $value_type)
    {
        $this->value_type = $value_type;
    }
}

File: test/java/DopTest/GsonTest/src/com/ffan/dop/MapItem.java
package com.ffan.dop;

import java.util.HashMap;
import java.util.Map;

class MapItem extends Item {
    /**
     * 所有项
     */
    private Map<Item, Item> items;

    /**
     * 构造函数
     */
    MapItem(int size) {
        this.type = ItemType.MAP_TYPE;
        this.items = new HashMap<Item, Item>(size);
    }
    
    /**
     * 获取值
     */
    public void add(Item key, Item value) {
        this.items.put(key, value);
    }

    /**
     * 获取所有项
     */
    public Map<Item, Item> getItems() {
        return this.items;
    }
}